platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -121,17 +125,20 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
 
         val options = DefaultInternalOptions(startParameter.systemPropertiesArgs)
         val isolatedProjects = startParameter.isolatedProjects.get()
+        val configurationCacheRequested = startParameter.configurationCache.get()
         val parallelProjectExecution = isolatedProjects || requirements.startParameter.isParallelProjectExecutionEnabled
         val parallelToolingActions = parallelProjectExecution && options.getOption(parallelBuilding).get()
         val invalidateCoupledProjects = isolatedProjects && options.getOption(invalidateCoupledProjects).get()
         val modelAsProjectDependency = isolatedProjects && options.getOption(modelProjectDependencies).get()
 
         return if (requirements.isCreatesModel) {
-            // When creating a model, disable certain features - only enable configure on demand and configuration cache when isolated projects is enabled
+            // When creating a model, disable certain features - only enable configure on demand when isolated projects is enabled
+            val configurationCacheForModels = configurationCacheRequested && options.getOption(configurationCacheForModels).get()
+            val configurationCache = isolatedProjects || configurationCacheForModels
             DefaultBuildModelParameters(
                 parallelProjectExecution = parallelProjectExecution,
                 configureOnDemand = false,
-                configurationCache = isolatedProjects,
+                configurationCache = configurationCache,
```

Comment:
This is the main change, untying the CC mode from Isolated Projects flag when creating models